nservicebus/operations/async-analyzer.md

```diff
@@ -0,0 +1,60 @@
+---
+title: Async Analyzer
+summary: How to use the async analyzer to avoid missing await operators
+component: Core
+versions: '[7,]'
+---
+
+Asynchronous APIs bring great performance improvements. However, they introduce new risks for bugs due to missed `await` operators. NServiceBus comes with a built-in Roslyn analyzer to detect those missing operators.
+
+## Functionality
+
+The analyzer will point out missing `await` operators when using asynchronous methods on the following classes and interfaces:
+* `IMessageHandlerContext`
+* `IMessageSession`
+* `Saga`
+* `IEndpointInstance`
+* `Endpoint`
+* `IUniformSession` (from the [UniformSession package](/nservicebus/messaging/uniformsession.md))
+
+## Limitations
+
+This analyzer inspects only the APIs contained on the classes and interfaces as described in the previous section. Other asynchronous operations on NServiceBus APIs are not analyzed.
+
+The analyzer checks for ignored `Task`s returned from the scanned methods. If the `Task` is assigned to a variable or passed to another method, the analyzer will not analyze this `Task`'s usage any further.
```

Comment:
Should we also add the fact that using any of the `Task`'s members will also prevent the diagnostic from reporting?